pkg/extensions/tbtc/recovery/derive_address_test.go

```diff
@@ -223,8 +225,18 @@ var resolveAddressData = map[string]struct {
 func TestResolveAddress(t *testing.T) {
 	for testName, testData := range resolveAddressData {
 		t.Run(testName, func(t *testing.T) {
+			dir, err := ioutil.TempDir("", "example")
+			if err != nil {
+				t.Fatal(err)
+			}
+			defer os.RemoveAll(dir)
+			dis, err := NewDerivationIndexStorage(dir)
+			if err != nil {
+				t.Fatal(err)
+			}
 			resolvedAddress, err := ResolveAddress(
 				testData.beneficiaryAddress,
+				dis,
```

Comment:
I don't have enough context but shouldn't this test fail now? There are no files in temp dir created so `ResolveAddress` will always use the first index while `deriveAddressTestData` uses a bunch of other address indexes.

Another option to do not use on-disk persistence here would be to define an interface or just accept a `GetIndex` function as a parameter to `ResolveAddress`, especially if we take into account when `Save` function should be called (see the previous comment).

Reply:
> I don't have enough context but shouldn't this test fail now?

The test data we were using was a single extended public key at `/0` (the default), and the rest were different forms of btc addresses that don't touch the persistence codepath.

I edited the test to be less implementation specific by providing a list of used indexes in each test case - should be good now!